command/kv_put.go

```diff
@@ -0,0 +1,249 @@
+package command
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/mitchellh/cli"
+)
+
+// KVPutCommand is a Command implementation that is used to write data to the
+// key-value store.
+type KVPutCommand struct {
+	Ui cli.Ui
+
+	// testStdin is the input for testing.
+	testStdin io.Reader
+}
+
+func (c *KVPutCommand) Help() string {
+	helpText := `
+Usage: consul kv put [options] KEY [DATA]
+
+  Writes the data to the given path in the key-value store. The data can be of
+  any type, but it will be transported as a base64-encoded string for safe
+  transport:
+
+      $ consul kv put config/redis/maxconns 5
+
+  The data can also be consumed from a file on disk by prefixing with the "@"
+  symbol. For example:
+
+      $ consul kv put config/program/license @license.lic
+
+  The DATA argument itself is optional. If omitted, this will create an empty
+  key-value pair at the specified path:
+
+      $ consul kv put webapp/beta/active
+
+  To perform a Check-And-Set operation, specify the -cas flag with the
+  appropriate -modify-index flag corresponding to the key you want to perform
+  the CAS operation on:
+
+      $ consul kv put -cas -modify-index=844 config/redis/maxconns 5
+
+  Additional flags and more advanced use cases are detailed below.
+
+` + apiOptsText + `
+
+KV Put Options:
+
+  -acquire                Obtain a lock on the key. If the key does not exist,
+                          this operation will create the key and obtain the
+                          lock. The session must already exist and be specified
+                          via the -session flag. The default value is false.
+
+  -cas                    Perform a Check-And-Set operation. If this value is
```

Comment:
Same as delete, if they CAS they must supply an index, otherwise it should fail.